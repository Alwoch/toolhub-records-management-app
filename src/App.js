import "./App.css";
import { Routes, Route } from "react-router-dom";

// components
import CardContainer from "./components/Dashboard/CardContainer";
import Leaderboard from "./components/Leaderboard/Leaderboard";
import Home from "./components/Home/Home";

function App() {
  return (
    <div className=".container-xxl">
      <Routes>
        <Route exact path="/leaderboard" element={<Leaderboard />} />
        <Route exact path="/" element={<Home/>} />
        <Route exact path="/dashboard" element={<CardContainer/>} />
      </Routes>
    </div>
  );
}

export default App;
