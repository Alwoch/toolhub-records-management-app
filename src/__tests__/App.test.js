import { render,screen} from '@testing-library/react';
import renderer from "react-test-renderer"
import { BrowserRouter } from "react-router-dom";
import App from '../App';

//Tests cases
  //App component should render correctly
  //rendered App matches the snapshot

test("should render App component successfully",()=>{
 render(<BrowserRouter><App/></BrowserRouter>);
 const AppElement=screen.getByTestId('app');
 expect(AppElement).toBeInTheDocument();
})

test("App should match snapshot",()=>{
 const AppTree=renderer.create(<BrowserRouter><App/></BrowserRouter>).toJSON()
 expect(AppTree).toMatchSnapshot()
})