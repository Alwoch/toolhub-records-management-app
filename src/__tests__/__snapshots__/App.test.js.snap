// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<div
  className=".container-xxl"
  data-testid="app"
>
  <nav
    className=".container-fluid"
    data-testid="nav"
  >
    <h3>
      Toolhub Records
    </h3>
    <div>
      <ul
        className="#nav-list"
        id="nav-list"
      >
        <li>
          <a
            className="active"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="/dashboard"
            onClick={[Function]}
          >
            Dashboard
          </a>
        </li>
        <li>
          <a
            href="/leaderboard"
            onClick={[Function]}
          >
            Leaderboard
          </a>
        </li>
      </ul>
    </div>
    <div
      id="mobile"
      onClick={[Function]}
    >
      <i
        className="fas fa-bars"
        id="bar"
      />
    </div>
  </nav>
  <section
    className=".container-xxl"
  >
    <div
      className="mb-3 input-group"
    >
      <input
        aria-describedby="basic-addon2"
        aria-label="search tool"
        className="form-control"
        placeholder="search tool"
      />
      <button
        className="button btn btn-outline-secondary"
        disabled={false}
        id="button-addon2"
        type="button"
      >
        search
      </button>
    </div>
    <p
      className="home-info"
    >
      The Wikimedia tool below is missing some important information. can you find it?
    </p>
    <p>
      Click 
      <span>
        I got this
      </span>
       to add this information or
       
      <span>
        Next
      </span>
       to see another tool
    </p>
    <div
      className="card"
    >
      <div
        className="card-header"
      >
        Tool:
      </div>
      <div
        className="card-body"
      >
        <div
          className="card-title h5"
        >
          Pywikibot
        </div>
        <p
          className="card-text"
        >
          api_url
        </p>
        <button
          className="btn btn-primary"
          disabled={false}
          type="button"
        >
          I Got this!
        </button>
        <button
          className="btn btn-primary"
          disabled={false}
          type="button"
        >
          Next Tool
        </button>
      </div>
    </div>
  </section>
</div>
`;
